chainer/functions/connection/n_step_rnn.py

```diff
@@ -0,0 +1,480 @@
+
+                # mini-batch size
+                h_type.shape[1] == c_type.shape[1],
+
+                # hidden size
+                h_type.shape[2] == c_type.shape[2],
+            )
+        else:
+            # RNN, GRU
+            (h_type, ), in_types = _split(in_types,
+                                          self.rnn_params['n_cell'])
+            type_check.expect(
+                h_type.dtype == numpy.float32,
+
+                h_type.ndim == 3,
+                h_type.shape[0] == self.n_layers * self.rnn_direction,
+            )
+
+        w_types, in_types = _split(in_types,
+                                   self.n_layers * self.rnn_direction *
+                                   self.rnn_params['n_W'])
+        b_types, in_types = _split(in_types,
+                                   self.n_layers * self.rnn_direction *
+                                   self.rnn_params['n_W'])
+        x_types = in_types
+        for x_type in x_types:
+            type_check.expect(
+                x_type.dtype == numpy.float32,
+                x_type.ndim == 2,
+            )
+        for x1_type, x2_type in zip(x_types, x_types[1:]):
+            type_check.expect(
+                # Check if xs are sorted by descending lengths
+                x1_type.shape[0] >= x2_type.shape[0],
+                x1_type.shape[1] == x2_type.shape[1])
+
+        in_size = x_types[0].shape[1]
+        out_size = h_type.shape[2]
+
+        for layer in six.moves.range(self.n_layers):
+            for i in six.moves.range(self.rnn_params['n_W']):
+                for di in six.moves.range(self.rnn_direction):
+                    ind = (layer * self.rnn_direction +
+                           di) * self.rnn_params['n_W'] + i
+                    w_type = w_types[ind]
+                    b_type = b_types[ind]
+                    if self.rnn_direction == 1:
+                        # Uni-direction
+                        if layer == 0 and i < (self.rnn_params['n_W'] / 2):
+                            w_in = in_size
+                        else:
+                            w_in = out_size
+                    else:
+                        # Bi-direction
+                        if layer == 0 and i < 3:
+                            w_in = in_size
+                        elif layer > 0 and i < 3:
+                            w_in = out_size * self.rnn_direction
+                        else:
+                            w_in = out_size
+
+                    type_check.expect(
+                        w_type.dtype == numpy.float32,
+                        w_type.ndim == 2,
+                        w_type.shape[0] == out_size,
+                        w_type.shape[1] == w_in,
+
+                        b_type.dtype == numpy.float32,
+                        b_type.ndim == 1,
+                        b_type.shape[0] == out_size,
+                    )
+
+    def forward(self, inputs):
+        if self.rnn_is_lstm_flag:
+            # LSTM
```

Comment:
Is it possible to make a function that prepares the descriptors related `c` and to overwrite the function in `NStepLSTM`? The spirit of this comment is same as that of l. 150.